src/_pytest/assertion/util.py

```diff
@@ -151,6 +151,8 @@ def isiterable(obj):
                 elif type(left) == type(right) and (isdatacls(left) or isattrs(left)):
                     type_fn = (isdatacls, isattrs)
                     explanation = _compare_eq_cls(left, right, verbose, type_fn)
+                elif verbose:
+                    explanation = _compare_eq_verbose(left, right)
```

Comment:
This uses it with `-v` already, no?

Reply:
Yes. That's also what happens in the case of e.g. strings: [https://github.com/pytest-dev/pytest/blob/master/src/_pytest/assertion/util.py#L176](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/assertion/util.py#L176)

The filtering based on -vv happens down the line somewhere else (in all cases).

So with -v this function will generate full output but pytest won't ultimately show the whole of it unless -vv is given.

Reply:
The test is showing the output of `call_equal` which calls these functions directly. Shortening (without -vv) takes place here I think: [https://github.com/pytest-dev/pytest/blob/master/src/_pytest/assertion/truncate.py](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/assertion/truncate.py)

That looks like a deliberate design and seems reasonable to me so I'm not sure what needs fixing.

Reply:
I think it makes sense that truncation happens later down the line, otherwise every implementation would need to implement or call truncation (IIUC).

Reply:
But yeah, I was also confused for this same reason, I had to look at the rest of the code to understand what was going on.

Reply:
@nicoddemus 
I think we should create a follow-up issue/PR for this.
Have you looked closer at it already?